src/solvers/refinement/string_refinement.cpp

```diff
@@ -896,6 +896,59 @@ exprt string_refinementt::substitute_array_with_expr(
   }
 }
 
+/// Pads an array represented by a list of with_expr by propagating values to
+/// the left.
+exprt pad_array_with_expr(const exprt &expr, std::size_t string_max_length)
+{
+  PRECONDITION(expr.type().id()==ID_array);
+  mp_integer max_index=0;
+  exprt it=expr;
+
+  while(it.id()==ID_with)
+  {
+    with_exprt with_expr=to_with_expr(it);
+    it=with_expr.old();
+    mp_integer index;
+    bool error=to_integer(to_constant_expr(with_expr.where()), index);
+    PRECONDITION(!error);
+    max_index=
+      (!error && index>max_index && max_index<string_max_length)?
+        index:max_index;
+  }
+  // We need a +1 because the maximal index is the length minus 1 and +1 to
+  // store the default value
+  std::vector<exprt> vector(max_index.to_long()+2);
+  std::set<std::size_t> initialized;
```

Comment:
Do we really need this, or could we just use a nil exprt to signify an unassigned entry in the vector?